src/terraform/structure/tf_taggable.go

```diff
@@ -1,679 +1,79 @@
-	"azurerm_virtual_network_gateway",
-	"azurerm_virtual_network_gateway_connection",
-	"azurerm_virtual_wan",
-	"azurerm_vmware_private_cloud",
-	"azurerm_vpn_gateway",
-	"azurerm_vpn_server_configuration",
-	"azurerm_vpn_site",
-	"azurerm_web_application_firewall_policy",
-	"azurerm_windows_virtual_machine",
-	"azurerm_windows_virtual_machine_scale_set",
-	"google_active_directory_domain",
-	"google_assured_workloads_workload",
-	"google_bigquery_dataset",
-	"google_bigquery_job",
-	"google_bigquery_table",
-	"google_bigtable_instance",
-	"google_cloud_identity_group",
-	"google_cloudfunctions_function",
-	"google_composer_environment",
-	"google_compute_disk",
-	"google_compute_image",
-	"google_compute_instance",
-	"google_compute_instance_from_template",
-	"google_compute_instance_template",
-	"google_compute_region_disk",
-	"google_compute_snapshot",
-	"google_dataflow_job",
-	"google_dataproc_cluster",
-	"google_dataproc_job",
-	"google_dataproc_workflow_template",
-	"google_dialogflow_cx_intent",
-	"google_dns_managed_zone",
-	"google_eventarc_trigger",
-	"google_filestore_instance",
-	"google_game_services_game_server_cluster",
-	"google_game_services_game_server_config",
-	"google_game_services_game_server_deployment",
-	"google_game_services_realm",
-	"google_gke_hub_membership",
-	"google_healthcare_consent_store",
-	"google_healthcare_dicom_store",
-	"google_healthcare_fhir_store",
-	"google_healthcare_hl7_v2_store",
-	"google_kms_crypto_key",
-	"google_memcache_instance",
-	"google_ml_engine_model",
-	"google_monitoring_notification_channel",
-	"google_network_management_connectivity_test",
-	"google_network_services_edge_cache_keyset",
-	"google_network_services_edge_cache_origin",
-	"google_network_services_edge_cache_service",
-	"google_notebooks_instance",
-	"google_privateca_ca_pool",
-	"google_privateca_certificate",
-	"google_privateca_certificate_authority",
-	"google_project",
-	"google_pubsub_subscription",
-	"google_pubsub_topic",
-	"google_redis_instance",
-	"google_secret_manager_secret",
-	"google_spanner_instance",
-	"google_storage_bucket",
-	"google_tpu_node",
-	"google_vertex_ai_dataset",
-	"google_workflows_workflow",
+import (
+	"strings"
+
+	"github.com/ahmetb/go-linq/v3"
+	tfjson "github.com/hashicorp/terraform-json"
+	awsv2 "github.com/lonegunmanb/terraform-aws-schema/v2/generated"
+	awsv3 "github.com/lonegunmanb/terraform-aws-schema/v3/generated"
+	awsv4 "github.com/lonegunmanb/terraform-aws-schema/v4/generated"
+	awsv5 "github.com/lonegunmanb/terraform-aws-schema/v5/generated"
+	azurev2 "github.com/lonegunmanb/terraform-azurerm-schema/v2/generated"
+	azurev3 "github.com/lonegunmanb/terraform-azurerm-schema/v3/generated"
+	googlev2 "github.com/lonegunmanb/terraform-google-schema/v2/generated"
+	googlev3 "github.com/lonegunmanb/terraform-google-schema/v3/generated"
+	googlev4 "github.com/lonegunmanb/terraform-google-schema/v4/generated"
+)
+
+var TfTaggableResourceTypes []string
+
+func init() {
```

Comment:
do you know how long the initialization takes? this will be recreated each time the CLI is used or am I missing something?

Reply:
I'll do a benchmark so you can make the call based on that. I was thinking about converting the json marshal operation into a lazy load, but it wasn't my priority before. If you're not satisfied with the benchmark result, then I'll make the load to lazy load.

Reply:
Hi @gruebel I just recalled that we cannot boost the `init` by lazy load since we need to filter out the resources that contains `tags`. Here's the benchmark on my laptop and if you're not satisfied with it, maybe we can boost the load by code generate?:

```text
GOROOT=C:\Users\hezijie\scoop\apps\go\current #gosetup
GOPATH=C:\Users\hezijie\go #gosetup
C:\Users\hezijie\scoop\apps\go\current\bin\go.exe test -c -o C:\Users\hezijie\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___BenchmarkInit_in_github_com_bridgecrewio_yor_src_terraform_structure.test.exe github.com/bridgecrewio/yor/src/terraform/structure #gosetup
C:\Users\hezijie\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___BenchmarkInit_in_github_com_bridgecrewio_yor_src_terraform_structure.test.exe -test.v -test.paniconexit0 -test.bench ^\QBenchmarkInit\E$ -test.run ^$ -test.count=100 #gosetup
goos: windows
goarch: amd64
pkg: github.com/bridgecrewio/yor/src/terraform/structure
cpu: 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
BenchmarkInit
BenchmarkInit-8              246           5656524 ns/op
BenchmarkInit-8              270           4959608 ns/op
BenchmarkInit-8              211           5061141 ns/op
BenchmarkInit-8              249           4825400 ns/op
BenchmarkInit-8              310           3281290 ns/op
BenchmarkInit-8              396           3308266 ns/op
BenchmarkInit-8              387           3507547 ns/op
BenchmarkInit-8              358           3308801 ns/op
BenchmarkInit-8              357           3374912 ns/op
BenchmarkInit-8              360           3287701 ns/op
BenchmarkInit-8              396           3366633 ns/op
BenchmarkInit-8              386           3320744 ns/op
BenchmarkInit-8              391           5482833 ns/op
BenchmarkInit-8              213           4862594 ns/op
BenchmarkInit-8              285           3868470 ns/op
BenchmarkInit-8              303           3673442 ns/op
BenchmarkInit-8              354           3305469 ns/op
BenchmarkInit-8              338           3350286 ns/op
BenchmarkInit-8              387           3335693 ns/op
BenchmarkInit-8              358           3304018 ns/op
BenchmarkInit-8              361           3436961 ns/op
BenchmarkInit-8              350           3917158 ns/op
BenchmarkInit-8              332           3217946 ns/op
BenchmarkInit-8              388           3363009 ns/op
BenchmarkInit-8              348           3279345 ns/op
BenchmarkInit-8              367           3206369 ns/op
BenchmarkInit-8              391           3257789 ns/op
BenchmarkInit-8              368           3270909 ns/op
BenchmarkInit-8              352           3235348 ns/op
BenchmarkInit-8              386           3352852 ns/op
BenchmarkInit-8              318           3525487 ns/op
BenchmarkInit-8              362           3526662 ns/op
BenchmarkInit-8              333           3532842 ns/op
BenchmarkInit-8              376           3321541 ns/op
BenchmarkInit-8              356           3308681 ns/op
BenchmarkInit-8              360           3678722 ns/op
BenchmarkInit-8              367           3250190 ns/op
BenchmarkInit-8              364           3266191 ns/op
BenchmarkInit-8              363           3474205 ns/op
BenchmarkInit-8              340           3352039 ns/op
BenchmarkInit-8              338           3202551 ns/op
BenchmarkInit-8              340           3348379 ns/op
BenchmarkInit-8              320           3288201 ns/op
BenchmarkInit-8              374           3283416 ns/op
BenchmarkInit-8              338           3218595 ns/op
BenchmarkInit-8              376           3346400 ns/op
BenchmarkInit-8              368           3320164 ns/op
BenchmarkInit-8              316           3336018 ns/op
BenchmarkInit-8              326           3993416 ns/op
BenchmarkInit-8              302           3397222 ns/op
BenchmarkInit-8              342           3320784 ns/op
BenchmarkInit-8              344           3186624 ns/op
BenchmarkInit-8              327           3182577 ns/op
BenchmarkInit-8              373           3342972 ns/op
BenchmarkInit-8              358           3381648 ns/op
BenchmarkInit-8              357           3348296 ns/op
BenchmarkInit-8              355           3211068 ns/op
BenchmarkInit-8              331           3260557 ns/op
BenchmarkInit-8              348           3390506 ns/op
BenchmarkInit-8              358           3988988 ns/op
BenchmarkInit-8              376           3296542 ns/op
BenchmarkInit-8              367           3203626 ns/op
BenchmarkInit-8              330           3174616 ns/op
BenchmarkInit-8              378           3210405 ns/op
BenchmarkInit-8              346           3271160 ns/op
BenchmarkInit-8              358           3351525 ns/op
BenchmarkInit-8              339           3178830 ns/op
BenchmarkInit-8              285           3530353 ns/op
BenchmarkInit-8              330           3334848 ns/op
BenchmarkInit-8              386           3372618 ns/op
BenchmarkInit-8              333           3274653 ns/op
BenchmarkInit-8              355           3671500 ns/op
BenchmarkInit-8              322           3359247 ns/op
BenchmarkInit-8              366           3307886 ns/op
BenchmarkInit-8              343           3992340 ns/op
BenchmarkInit-8              358           3436902 ns/op
BenchmarkInit-8              384           3342238 ns/op
BenchmarkInit-8              342           3323672 ns/op
BenchmarkInit-8              357           3265490 ns/op
BenchmarkInit-8              372           3465086 ns/op
BenchmarkInit-8              380           3311641 ns/op
BenchmarkInit-8              333           3364876 ns/op
BenchmarkInit-8              369           3275553 ns/op
BenchmarkInit-8              393           3413220 ns/op
BenchmarkInit-8              310           3503894 ns/op
BenchmarkInit-8              331           3305068 ns/op
BenchmarkInit-8              393           3400776 ns/op
BenchmarkInit-8              357           3284698 ns/op
BenchmarkInit-8              364           4481733 ns/op
BenchmarkInit-8              307           3947393 ns/op
BenchmarkInit-8              333           3271755 ns/op
BenchmarkInit-8              241           6217576 ns/op
BenchmarkInit-8              361           3335206 ns/op
BenchmarkInit-8              343           3278905 ns/op
BenchmarkInit-8              355           3612213 ns/op
BenchmarkInit-8              348           3298262 ns/op
BenchmarkInit-8              390           3479846 ns/op
BenchmarkInit-8              406           3255017 ns/op
BenchmarkInit-8              355           3320015 ns/op
BenchmarkInit-8              349           3255885 ns/op
PASS

Process finished with the exit code 0
```